plugins/inventory/vultr.py

```diff
@@ -180,22 +193,48 @@ def parse(self, inventory, loader, path, cache=True):
 
             server = Vultr.normalize_result(server, SCHEMA)
 
-            self.inventory.add_host(host=server['name'], group='vultr')
+            host_name = server['name']
+            if hostname_preference != 'name':
+                host_name = server[hostname_preference]
 
-            for attribute, value in server.items():
-                self.inventory.set_variable(server['name'], attribute, value)
+            host_vars = {}
+            for k, v in server.items():
+                host_vars[k] = v
 
-            if hostname_preference != 'name':
-                self.inventory.set_variable(server['name'], 'ansible_host', server[hostname_preference])
+            if not self._passes_filters(host_filters, host_vars, host_name, strict):
+                self.display.vvv("Host {0} did not pass all filters".format(server['name']))
+                continue
+
+            self.inventory.add_host(host=host_name, group='vultr')
+
+            for attribute, value in server.items():
+                self.inventory.set_variable(host_name, attribute, value)
 
             # Use constructed if applicable
             strict = self.get_option('strict')
```

Comment:
Oh, I think I have clashed the `strict` and `strict_filtering` options, sorry.

Presumably we should just use `strict`, and pass that to the jinja processing?